esp/services/ws_smc/ws_smcService.cpp

```diff
@@ -2255,9 +2242,8 @@ void CWsSMCEx::getStatusServerInfo(IEspContext &context, const char *serverType,
     if (!serverType || !*serverType)
         throw MakeStringException(ECLWATCH_MISSING_PARAMS, "Server type not specified.");
 
-    Owned<CActivityInfo> activityInfo = getActivityInfo(context);
-    if (!activityInfo)
-        throw MakeStringException(ECLWATCH_INTERNAL_ERROR, "Failed to get Activity Info cache.");
+    Owned<CActivityInfo> activityInfo = activityInfoReader->getActivityInfo();
+    assert(activityInfo);
```

Comment:
Should use assertex instead,
*but*, It could legitimately be null here, if stopping before ever got started, so you should not assert..

Reply:
Will throw an exception to inform ESP client that Activity Info is not available at this time.